pkg/domain/infra/abi/system.go

```diff
@@ -61,21 +61,70 @@ func (ic *ContainerEngine) Info(ctx context.Context) (*define.Info, error) {
 	return info, nil
 }
 
+// BuildPrune removes stage build containers, dangling images, all build cache data.
+// Note: This is not safe operation and should be executed only when no builds are in progress. It can interfere with builds in process.
+func (ic *ContainerEngine) BuildPrune(ctx context.Context, options entities.SystemPruneOptions) (*entities.SystemPruneReport, error) {
+	var systemPruneReport = new(entities.SystemPruneReport)
+
+	if options.All || options.Volume || len(options.Filters) > 0 || options.External {
+		return nil, fmt.Errorf("system prune --build cannot be combined with other options")
+	}
+
+	reclaimedSpace := (uint64)(0)
+	found := true
+	for found {
+		found = false
+
+		// Prune Stage Containers
+		stageContainersPruneReports, err := ic.Libpod.PruneStageContainers()
+		if err != nil {
+			return nil, err
+		}
+		if len(stageContainersPruneReports) > 0 {
+			found = true
+		}
+		reclaimedSpace += reports.PruneReportsSize(stageContainersPruneReports)
+		systemPruneReport.ContainerPruneReports = append(systemPruneReport.ContainerPruneReports, stageContainersPruneReports...)
+
+		// Prune Images
+		imagePruneOptions := entities.ImagePruneOptions{
+			External:   true,
+			BuildCache: true,
+		}
+		imageEngine := ImageEngine{Libpod: ic.Libpod}
+		imagePruneReports, err := imageEngine.Prune(ctx, imagePruneOptions)
+		if err != nil {
+			return nil, err
+		}
+		if len(imagePruneReports) > 0 {
+			found = true
+		}
+		reclaimedSpace += reports.PruneReportsSize(imagePruneReports)
+		systemPruneReport.ImagePruneReports = append(systemPruneReport.ImagePruneReports, imagePruneReports...)
```

Comment:
This seems inconsistent with the documentation, from the docs you wrote it sounds like it only prune the build containers but this prunes everything other containers and images as well.
I don't think this is desirable.
If we want that behavior then the code should not cause a conflict with --build option and only do this in addition to the existing code in SystemPrune() instead of duplicating the image logic here.